### ISSUE TITLE
:jack_o_lantern: Hacktoberfest 2022: More snapshot tests

### ISSUE THREAD

**Issue by spenserblack** [spenserblack/steamdown#59]
The purpose of this issue is to not only make this app more well-tested, but introduce you to automated testing, jest, and snapshot tests.

******

The main package can use more snapshot tests. Check out https://github.com/spenserblack/steamdown/blob/main/CONTRIBUTING.md#updating-steamdown for some docs on how to manage snapshot tests and useful links.

**Comment by shivam-Purohit** [comment]
Hey can you elaborate a little more about the issue, I can learn and work on this if you like.

**Comment by spenserblack** [comment]
Sure! Basically, right now there is only [one Markdown file (`basic.md`)](https://github.com/spenserblack/steamdown/blob/main/packages/steamdown/__tests__/input/basic.md) that gets a snapshot test here: https://github.com/spenserblack/steamdown/blob/60348665d351b9a91bc4c72e6f39b510c0858e3b/packages/steamdown/__tests__/snapshots.spec.ts#L7-L12

These snapshot tests can be used to guarantee the stability of outputs. As long as the snapshot test passes, you know the output hasn't changed for the input. The snapshots are [here](https://github.com/spenserblack/steamdown/blob/60348665d351b9a91bc4c72e6f39b510c0858e3b/packages/steamdown/__tests__/__snapshots__/snapshots.spec.ts.snap). You don't need to worry about creating snapshots -- they will be created automatically for new tests.

Basically, all you need to do is
1. create a new Markdown file with some useful test scenarios in [`packages/steamdown/__tests__/input`](https://github.com/spenserblack/steamdown/tree/main/packages/steamdown/__tests__/input)
2. Register that test file by adding its filename (without the `.md` extension) on [this line](https://github.com/spenserblack/steamdown/blob/60348665d351b9a91bc4c72e6f39b510c0858e3b/packages/steamdown/__tests__/snapshots.spec.ts#L7)
